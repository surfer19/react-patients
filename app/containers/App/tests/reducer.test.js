import globalReducer from '../reducer';
import {
  loadRecords,
  loadRecordsSuccess,
  loadRecordsError,
  setPractitioner,
  setPractitionerSuccess,
  setPatientId,
  setPatientIdSuccess,
} from '../actions';

describe('globalReducer', () => {
  const id = '123456';
  const mockedRecords = [
    {
      patientId: '5cc17bd0a75e3173228f1684',
      practitionerId: '2588ac7f57fd9b49',
      name: 'Wilson',
      surname: 'Wade',
    },
    {
      patientId: '5cc17bd0a5283537f6c20a4a',
      practitionerId: 'a7e46537714eca83',
      name: 'Bridges',
      surname: 'Odom',
    },
  ];
  it('returns the initial state', () => {
    expect(globalReducer(undefined, {})).toMatchSnapshot();
  });

  it('handles the loadRecords action', () => {
    expect(globalReducer({}, loadRecords())).toMatchSnapshot();
  });

  it('handles the loadRecordsSuccess action', () => {
    expect(
      globalReducer({}, loadRecordsSuccess(mockedRecords)),
    ).toMatchSnapshot();
  });

  it('handles the loadRecordsError action', () => {
    expect(globalReducer({}, loadRecordsError())).toMatchSnapshot();
  });

  it('handles the setPractitioner action', () => {
    expect(globalReducer({}, setPractitioner(id))).toMatchSnapshot();
  });

  it('handles the setPractitionerSuccess action', () => {
    expect(globalReducer({}, setPractitionerSuccess(id))).toMatchSnapshot();
  });

  it('handles the setPatientId action', () => {
    expect(globalReducer({}, setPatientId(id))).toMatchSnapshot();
  });

  it('handles the setPatientIdSuccess action', () => {
    expect(globalReducer({}, setPatientIdSuccess(id))).toMatchSnapshot();
  });
});
