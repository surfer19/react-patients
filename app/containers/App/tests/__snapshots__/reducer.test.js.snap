// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`globalReducer handles the loadRecords action 1`] = `Object {}`;

exports[`globalReducer handles the loadRecordsError action 1`] = `Object {}`;

exports[`globalReducer handles the loadRecordsSuccess action 1`] = `
Object {
  "patientsRecords": Array [
    Object {
      "name": "Wilson",
      "patientId": "5cc17bd0a75e3173228f1684",
      "practitionerId": "2588ac7f57fd9b49",
      "surname": "Wade",
    },
    Object {
      "name": "Bridges",
      "patientId": "5cc17bd0a5283537f6c20a4a",
      "practitionerId": "a7e46537714eca83",
      "surname": "Odom",
    },
  ],
}
`;

exports[`globalReducer handles the setPatientId action 1`] = `
Object {
  "patientId": "123456",
}
`;

exports[`globalReducer handles the setPatientIdSuccess action 1`] = `
Object {
  "patientId": "123456",
}
`;

exports[`globalReducer handles the setPractitioner action 1`] = `
Object {
  "practitionerId": "123456",
}
`;

exports[`globalReducer handles the setPractitionerSuccess action 1`] = `
Object {
  "practitionerId": "123456",
}
`;

exports[`globalReducer returns the initial state 1`] = `
Object {
  "patientId": "",
  "patientsRecords": Array [],
  "practitionerId": "",
}
`;
